bin/logstash.lib.sh

```diff
@@ -1,5 +1,29 @@
 unset CDPATH
-LOGSTASH_HOME=$(cd `dirname $0`/..; pwd)
+# This unwieldy bit of scripting is to try to catch instances where Logstash
+# was launched from a symlink, rather than a full path to the Logstash binary
+if [ -L $0 ]; then
```

Comment:
In other words, you'd never be able to find the util lib if you were starting with a symlink, which is the whole reason for this code's existence.  It's self-defeating to use a lib if the lib has to be found to be used :smile: